service/controller/app/v1/resource/chart/resource.go

```diff
@@ -113,3 +89,19 @@ func New(config Config) (*Resource, error) {
 func (r *Resource) Name() string {
 	return Name
 }
+
+// equals asseses the equality of ReleaseStates with regards to distinguishing fields.
+func equals(a, b v1alpha1.Chart) bool {
+	if a.Name != b.Name {
+		return false
+	}
+	if !reflect.DeepEqual(a.Spec, b.Spec) {
+		return false
+	}
```

Comment:
This is good but we should also compare ObjectMeta. As we may add or change the metadata in the future.

Reply:
But `ObjectMeta` contains several TimeStamp object it could be different in many cases. Can we select some few variables in ObjectMeta, just to compare?

Reply:
Good point. What about if we just consider labels and annotations?

Reply:
Sure, I would make that change right away.